cli/src/main/java/org/infinispan/cli/commands/kubernetes/Shell.java

```diff
@@ -45,53 +55,68 @@ public boolean isHelp() {
 
    @Override
    public CommandResult exec(ContextAwareCommandInvocation invocation) {
-      KubernetesClient client = ((KubernetesContextImpl)invocation.getContext()).getKubernetesClient();
+      KubernetesClient client = KubernetesContext.getClient(invocation);
       namespace = Kube.getNamespaceOrDefault(client, namespace);
-      if (podName == null) {
-         List<Pod> pods = client.pods().inNamespace(namespace).withLabel("infinispan_cr", name).list().getItems();
-         for(Pod pod : pods) {
-            PodStatus status = pod.getStatus();
-            if ("Running".equalsIgnoreCase(status.getPhase())) {
-               podName = pod.getMetadata().getName();
-               break;
-            }
-         }
-      }
-      if (podName == null) {
-         System.err.printf("No running pods available in service %s\n", name);
-      }
-
-      LatchListener latchListener = new LatchListener();
-      client.pods().inNamespace(namespace).withName(podName)
-            .readingInput(System.in)
-            .writingOutput(System.out)
-            .writingError(System.err)
-            .withTTY()
-            .usingListener(latchListener)
-            .exec("/opt/infinispan/bin/cli.sh");
-
-      try {
-         latchListener.latch.await();
-      } catch (InterruptedException e) {
+      GenericKubernetesResource infinispan = client.genericKubernetesResources(INFINISPAN_CLUSTER_CRD).inNamespace(namespace).withName(name).get();
+      if (infinispan == null) {
+         System.err.printf("No service '%s' in namespace '%s'\n", name, namespace);
```

Comment:
Shouldn't we throw an exception defined in `Messages.java` here?

Reply:
Done